File: Checkers/Assets/Scripts/Piece.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Piece : MonoBehaviour
{
    public bool IsKing { get; set; }
    public bool IsWhite { get; set; }

    public Piece(bool white)
    {
        IsWhite = white;
        IsKing = false;
    }

    public Piece(Piece p)
    {
        IsWhite = p.IsWhite;
        IsKing = p.IsKing;
    }

}

File: CheckersPythonEngine/game_engine/Piece.py
class Piece(object):
    isKing: bool
    isWhite: bool

    def __init__(self, white: bool, isKing: bool = False):
        self.isWhite = white
        self.isKing = isKing

    def __str__(self):
        return f"Piece(white={self.isWhite}, king={self.isKing})"

    def reverse_color(self):
        self.isWhite = not self.isWhite
